packages/url/src/test/index.test.js

```diff
@@ -30,28 +30,30 @@ import {
 } from '../';
 
 describe( 'isURL', () => {
-	it( 'returns true when given things that look like a URL', () => {
-		const urls = [
-			'http://wordpress.org',
-			'https://wordpress.org',
-			'HTTPS://WORDPRESS.ORG',
-			'https://wordpress.org/foo#bar',
-			'https://localhost/foo#bar',
-		];
-
-		expect( every( urls, isURL ) ).toBe( true );
-	} );
-
-	it( 'returns false when given things that don\'t look like a URL', () => {
-		const urls = [
-			'HTTP: HyperText Transfer Protocol',
-			'URLs begin with a http:// prefix',
-			'Go here: http://wordpress.org',
-			'http://',
-			'',
-		];
-
-		expect( every( urls, isURL ) ).toBe( false );
+	it.each( [
```

Comment:
Do we actually need to test all the possibilities here? I mean we're essentially testing the `URL` constructor. Just wondering what's the benefit of doing it.

Reply:
>I mean we're essentially testing the `URL` constructor. 

As far as the tests are concerned, it shouldn't care how the implementation chooses to validate the URL. The contract is that the string be valid per the URL Standard, not necessarily anything about the `URL` constructor itself. I grant that since the `URL` constructor and the specification are so closely tied, the difference here is pretty minimal. One advantage of having the tests is that it could support some potential future refactoring where we choose not to implement it using the `URL` constructor. It's hard to say whether it would ever make sense to undertake such a refactor, but it would be perfectly valid to do so.

Reply:
> As far as the tests are concerned, it shouldn't care how the implementation

Yes I completely agree here. We should be able to swap out `new URL` for `new AmazingNewURLAPIThatDaveWrote()` and the tests shouldn't be aware of the change. They just care whether it passes or not.

Reply:
We predicted the future quite well in this conversation chain 😉 (See #20172)

Reply:
> Do we actually need to test all the possibilities here? I mean we're essentially testing the `URL` constructor. Just wondering what's the benefit of doing it.

@WunderBart You may also be interested: #20537